src/coreclr/src/jit/morph.cpp

```diff
@@ -13678,8 +13688,9 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                     }
                     else if (temp->OperIsLocal())
                     {
-                        unsigned   lclNum = temp->AsLclVarCommon()->GetLclNum();
-                        LclVarDsc* varDsc = &lvaTable[lclNum];
+                        unsigned   lclNum  = temp->AsLclVarCommon()->GetLclNum();
```

Comment:
It is still confusing that we are checking parent struct type for local fields even if these checks never pass and the code is working because of that.

I would suggest separate this `else if (temp->OperIsLocal())`
as
```
else if (temp->OperIs(GT_LCL_VAR)) { the new logic}
else if  (temp->OperIs(GT_LCL_FLD)) // the old logic that was applying to `LCL_FLD` only.
{
    // Assumes that when Lookup returns "false" it will leave "fieldSeq" unmodified (i.e.
    // nullptr)
    assert(fieldSeq == nullptr);
    bool b = GetZeroOffsetFieldMap()->Lookup(op1, &fieldSeq);
    assert(b || fieldSeq == nullptr);

    if (fieldSeq != nullptr) // that is from the initial CoreCLR commit, no idea why we do it only
                             // when we have a zero offset.
    {
        // Append the field sequence, change the type.
        temp->AsLclFld()->SetFieldSeq(
            GetFieldSeqStore()->Append(temp->AsLclFld()->GetFieldSeq(), fieldSeq));
        temp->gtType = typ;

        foldAndReturnTemp = true;
    }
} 
```

Reply:
Look I think that we can address any other change as a follow up.

Reply:
I would like to simply check this in as it is today, without making any new changes.

Reply:
I'm not sure why there is a rush to check this in today; I think it's worth making the change @sandreenko suggests, as it doesn't seem reasonable to use the parent type in the `GT_LCL_FLD' case.

Reply:
I have made this change:

```
else if  (temp->OperIs(GT_LCL_FLD)) // the old logic that was applying to `LCL_FLD` only.
else if (temp->OperIs(GT_LCL_VAR)) { the new logic}
```

Reply:
I ended up changing it to this:

```
                    else if (temp->OperIsLocal())
                    {
                        if (temp->OperIs(GT_LCL_VAR))
                        {  // newer code }
                    }
                        if (!foldAndReturnTemp)
                        {  .// old code }
```